src/NHibernate/Linq/Visitors/NhPartialEvaluatingExpressionVisitor.cs

```diff
@@ -155,6 +158,42 @@ private Expression EvaluateSubtree(Expression subtree)
 
 		#region NH additions
 
+		protected override Expression VisitMethodCall(MethodCallExpression node)
+		{
+			DetectEvaluatableExpressionOnCollectionContains(node);
+			return base.VisitMethodCall(node);
+		}
+
+		private void DetectEvaluatableExpressionOnCollectionContains(MethodCallExpression expression)
+		{
+			if (!expression.Method.IsGenericMethod || ContainsMethodInfo != expression.Method.GetGenericMethodDefinition())
+				return;
+			var argument = expression.Arguments[0];
+			if (argument.NodeType != ExpressionType.Call)
+				return;
+
+			if(TryGetCollectionParameter((MethodCallExpression)argument))
+				_partialEvaluationInfo.AddEvaluatableExpression(argument);
```

Comment:
Here you're modifying external dependency, which you should not.

Reply:
I don't get it. Why I shouldn't touch it? It's just a wrapper around set of expressions that needs evaluation.  I can store it in separate field (as I did initially) - but to minimize change noise I used already present class for storing evaluatable expressions.

Reply:
> I don't get it. Why I shouldn't touch it?

Because this is the information provided by another concerning party. We already let this logic spread across solution that we trying to catch bugs doing 10th bugfix release.

Reply:
Well.. Ok... It can be stored separately. But I still think it's more proper to store it in `_partialEvaluationInfo` - external or not, it stores list of expressions that needs evaluation. So it's better to fill it also with our expressions - so all other external parties can handle it appropriately as evaluatable expression.

But before making further changes -  do you agree with this fix on principle?